### ISSUE TITLE
Optimize UAI Reader

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#486]
UAI reader takes too much time in reading files. Can test on this file:
[hailfinder.uai.txt](https://github.com/pgmpy/pgmpy/files/8246/hailfinder.uai.txt)

**Comment by khalibartan** [comment]
Will writing the module in cython will help.

**Comment by ankurankan** [comment]
@khalibartan I don't think we need to cythonize it. There should be a lot of scope for optimization in general.
